### DISCUSSION TITLE
Reading Smart Tables

### DISCUSSION THREAD

**Discussion by DarTas1** [abap2xlsx/abap2xlsx#1211] [Q&A]
Hello,
looking for a way to read smart tables using abap2xlsx, but could not identify a way, since zcl_excel_reader loads the ZCL_EXCEL_WORKSHEET object without any information about tables. Tried to upload a generated version from ZDEMO_EXCEL3 and read it, but no info on the generated table was found in the runtime. Is there actually no possibility to read these tables from imported excel file or am I doing something wrong?
Code snippet with which I tried to check it:

```abap
SELECTION-SCREEN BEGIN OF BLOCK blx WITH FRAME.
PARAMETERS: p_upfile TYPE string LOWER CASE MEMORY ID gr8.
SELECTION-SCREEN END OF BLOCK blx.

DATA: excel          TYPE REF TO zcl_excel,
      reader         TYPE REF TO zif_excel_reader,
      worksheet      TYPE REF TO zcl_excel_worksheet,
      lt_files       TYPE filetable,
      lv_rc          TYPE i,
      lv_file_filter TYPE string.

lv_file_filter = 'Excel Files (*.XLSX;*.XLSM)|*.XLSX;*.XLSM'.
cl_gui_frontend_services=>file_open_dialog( EXPORTING
                                              default_filename        = p_upfile
                                              file_filter             = lv_file_filter
                                            CHANGING
                                              file_table              = lt_files
                                              rc                      = lv_rc
                                            EXCEPTIONS
                                              OTHERS                  = 1 ).
READ TABLE lt_files INDEX 1 INTO p_upfile.

CREATE OBJECT reader TYPE zcl_excel_reader_2007.
excel = reader->load_file( i_filename = p_upfile ).

worksheet = excel->get_active_worksheet( ).

write: 'size of tables ', worksheet->get_tables_size( ).
```

Result is always 0
![image](https://github.com/abap2xlsx/abap2xlsx/assets/75167233/a86c9477-70b5-4eeb-a057-d646f2fea633)
Used the file:
[generated table.xlsx](https://github.com/abap2xlsx/abap2xlsx/files/15024862/generated.table.xlsx)

BR,
Dartas1

**Comment by sandraros** [comment]
Thanks. It's a missing feature, tables are only handled in the writer, they are not handled in the reader.
Could you [Create issue from discussion](https://github.com/abap2xlsx/abap2xlsx/issues/new?created_from_discussion_number=1211)?

**Reply by DarTas1** [reply]
Yep. Thanks

**Comment by sandraros** [comment]
In fact, sorry, this was fixed by #1159 (issue #1158).
